apiserver/facades/agent/provisioner/provisioner.go

```diff
@@ -1095,27 +1083,14 @@ func (h *prepareOrGetHandler) ProcessOneContainer(
 		}
 	}
 
-	// We do not ask the provider to allocate addresses for manually provisioned
-	// machines as we do not expect such machines to be recognised (LP:1796106).
-	var askProviderForAddress bool
-	if !hostIsManual {
-		askProviderForAddress = environs.SupportsContainerAddresses(ctx, env)
-	}
-
-	allocatedInfo := preparedInfo
-	if askProviderForAddress {
-		guestMachineTag := names.NewMachineTag(guestMachineName.String())
-		// supportContainerAddresses already checks that we can cast to an environ.Networking
-		networking := env.(environs.Networking)
-		var err error
-		allocatedInfo, err = networking.AllocateContainerAddresses(
-			ctx, hostInstanceID, guestMachineTag, preparedInfo)
-		if err != nil {
-			return errors.Trace(err)
-		}
-		h.logger.Debugf(ctx, "got allocated info from provider: %+v", allocatedInfo)
-	} else {
+	allocatedInfo, err := networkService.AllocateContainerAddresses(
+		ctx, hostInstanceID, guestMachineName.String(), preparedInfo)
+	if err != nil && !errors.Is(err, networkerrors.ContainerAddressesNotSupported) {
+		return errors.Trace(err)
+	} else if errors.Is(err, networkerrors.ContainerAddressesNotSupported) {
 		h.logger.Debugf(ctx, "using dhcp allocated addresses")
+	} else {
+		h.logger.Debugf(ctx, "got allocated info from provider: %+v", allocatedInfo)
 	}
```

Comment:
It is OK to chain error handling like this:
```suggestion
	if errors.Is(err, networkerrors.ContainerAddressesNotSupported) {
		h.logger.Debugf(ctx, "using dhcp allocated addresses")
	} else if err != nil {
		return errors.Trace(err)
	} else {
		h.logger.Debugf(ctx, "got allocated info from provider: %+v", allocatedInfo)
	}
```
I'd argue this is clearer and easier to read.